src/modules/mavlink/streams/ADSB_VEHICLE.hpp

```diff
@@ -103,6 +128,89 @@ class MavlinkStreamADSBVehicle : public MavlinkStream
 			sent = true;
 		}
 
+#if CONFIG_DRIVERS_TRANSPONDER_UAVIONIX
+
+		if (_mavlink->get_mode() == Mavlink::MAVLINK_MODE_UAVIONIX) {
+
+			// Required update for dynamic message is 5 [Hz]
+			mavlink_uavionix_adsb_out_dynamic_t dynamic_msg = {
+				.utcTime = UINT32_MAX,
+				.gpsLat = INT32_MAX,
+				.gpsLon = INT32_MAX,
+				.gpsAlt = INT32_MAX,
+				.baroAltMSL = INT32_MAX,
+				.accuracyHor = UINT32_MAX,
+				.accuracyVert = UINT16_MAX,
+				.accuracyVel = UINT16_MAX,
+				.velVert = INT16_MAX,
+				.velNS = INT16_MAX,
+				.VelEW = INT16_MAX,
+				.state = UAVIONIX_ADSB_OUT_DYNAMIC_STATE_ON_GROUND,
+				.squawk = static_cast<uint16_t>(_adsb_squawk.get()),
+				.gpsFix = 0,
+				.numSats = UINT8_MAX,
+				.emergencyStatus = static_cast<uint8_t>(_adsb_emergc.get())
+			};
+
+			vehicle_status_s vehicle_status;
+
+			if (_vehicle_status_sub.update(&vehicle_status)) {
+				if (vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_ARMED) {
+					dynamic_msg.state |= ~UAVIONIX_ADSB_OUT_DYNAMIC_STATE_ON_GROUND;
+				}
+			}
+
+			sensor_gps_s vehicle_gps_position;
+
+			if (_vehicle_gps_position_sub.update(&vehicle_gps_position)) {
```

Comment:
copy regardless of updated or only do the work and send the struct if updated

Reply:
I'll change it to copy regardless of the update since it expects an update at a 5 Hz rate.